### PR TITLE
Add Disabled State to List View

### PR DESCRIPTION
**Pull request by yih-wang**
This pr bases on the issue #313

### CODE REVIEW

**Comment by yih-wang** [comment]
#313

**Comment by yih-wang** [comment]
@mcarrano @LHinson  I use a light gray here for disabled rows to distinguish from the new
selected row highlighting (as it was mentioned in [#676](https://github.com/patternfly/patternfly/issues/676#issuecomment-312669489) that the two
colors we have implemented are too close to tell from each other). Please have a review.

File: pattern-library/content-views/list-view/design/design.md [hunk 1]
```diff
@@ -15,6 +15,8 @@
     - Content should wrap, displaying multiple rows of data
     - Content could be eliminated
 
+1. **Disabled Row:** Rows that are currently not available will be disabled. The cursor should appear visually disabled when user hovers over the row.
+
```
Comment (mcarrano) [inline pattern-library/content-views/list-view/design/design.md]: There seems to be a problem with the numbering for the items added.

**Comment by yih-wang** [comment]
@mcarrano I tried to correct the numbers but it didn't work. Seems like it's the bug in markdown because I found the same issue in the previous pr: https://github.com/LHinson/patternfly-design/blob/3047a787cf530337147fbb681c1ae3269655304e/pattern-library/content-views/list-view/design/design.md The number will count from 1 if it comes after a bulleted list.

**Comment by serenamarie125** [comment]
@yih-wang is it fair to assume that if something is disabled and has a checkbox, the checkbox would still be enabled, for example so the user could take an action to enable it?

Also curious as to what an expanded list view item looks like when it is expanded.  I'm guessing this may be an additional requirement, thoughts @mcarrano ?

**Comment by yih-wang** [comment]
@catrobson In my opinion, if user can still visit detailed data when hovering over the graph, then it shouldn't be grayed out. Similarly to a link or other elements which can be drilled down or visit, if we still allow users to view them (just forbidden to take actions to the whole row), they also shouldn't be grayed out. What's your thoughts?

**Comment by yih-wang** [comment]
@serenamarie125 Yes, we can directly enable the checkboxes in the disabled rows. But it may be not so safe that user can click it by accident (because they can't distinguish the checkboxes of disabled rows and enabled rows very well). To avoid this, we can set a button like "Enable Disabled Rows". When user clicking the button, the checkboxes of disabled rows change to active and other checkboxes are temporarily disabled (to avoid clicking by accident). After user choosing the rows they want to enable, the button change to "Confirm to Enable" or something. Click it will enable the rows. What's your thoughts?

**Comment by catrobson** [comment]
@yih-wang I agree if you can take action on the content, the content shouldn't be grayed out.  I would expect maybe there are different "levels" of disabled then:
1.  No action can be taken on the row:  Shouldn't just the checkbox and kabob be disabled in this case?  If drag and drop is enabled then a disabled state for that would be needed in this case too.
2. No action can be taken on the content of the row, but actions are allowed to the row itself:  All content showed as disabled, but the checkbox/kabob might be enabled?  This one is odd to me, I'm not sure what the use cases are for this one.
3. No action can be taken on the content or the row:  Would appear as designed for your design, everything in the row is disabled.

What do you think?

**Comment by mcarrano** [comment]
After reading this comment thread, I wonder if we should take a step back. What do we really mean when we disable a row?  Agree with @catrobson that if what we are saying is that actions cannot be taken on this object, why don't we just disable the kabob and/or specific actions?  Is it the row that's disabled or really the object referenced by that row? In applications I've worked on, where an object is disabled, that's reflected somehow by a status.  But any actions, selections, or drill-downs can still be valid on that object. Is this what we are trying to reflect by showing a disabled row or something else?

**Comment by yih-wang** [comment]
@mcarrano @catrobson  I agree with the partly disabled solution. But I'm confused about what we want to reflect through a disabled row. When I looked back into the original issue ([patternfly/patternfly#676](https://github.com/patternfly/patternfly/issues/676)), the reason we want to support the ability to disable a row is Angular PatternFly has the attribute. And that's why I created a mockup pretty consistent with the current Angular implementation. As there are some new points coming up in the discussion, I think It will be helpful if you can provide some use cases for reference. Thanks!

**Comment by Mgranfie** [comment]
I had been thinking about this use case when I reviewed the design as well
and went back and forth on how and if this would be something to convey as
a use case.

After reading these discussions, was thinking that maybe identifying the
status of the row:

A row represents an object, an object has states that are triggered by
conditions.
there are two levels of functionality, the list level functionality and the
object level functionality.

When an object is not accessible, the object level representation of it, in
a row, should reflect this by presenting as disabled.
The list level functionality may or may not present as disabled, depending
on the use case. For example, the object in the list may not be accessible
or I may not have rights to open and edit it, but I still may have the
right to select the object and delete it from my list.

So then do we provide only the presentation of the states that could occur,
knowing that there can be operational conditions that will cause the
row/object to enable or disable, or that there may be commands that the
user manually invokes, to change the status of objects in the row, just
inform around the flexibility of the control to address these use cases.

What if we inform that a row, which represents an object, is supported in
our lists and here is how it presents.... the application determines what
triggers the enabled and disabled states.

Not sure if this is helpful or more confusing?....

**Comment by mcarrano** [comment]
@yih-wang We discussed this issue during today's UXS-PF meeting and decided that the requirements must be further clarified before proceeding. Please put this on hold for now and I will create a new story once we reach agreement on what's needed.

**Comment by yih-wang** [comment]
@Mgranfie Thanks for your comment! It did give me some inspirations. It's a good point to separate the row functionality into two levels. As the issue was put on hold, I may not make any updates for the moment. But I'll take your suggestion into consideration when the story reopening again.

File: pattern-library/content-views/list-view/design/design.md [hunk 2]
```diff
@@ -14,6 +14,8 @@
   - When the viewport size is decreased:
     - Content should wrap, displaying multiple rows of data
     - Content could be eliminated
+    
+1. **Disabled Row:** Rows that are currently not available will be disabled. The cursor should appear visually disabled when user hovers over the row.
```
Comment (mcarrano) [inline pattern-library/content-views/list-view/design/design.md]: Please mark this feature as Optional

**Comment by mcarrano** [comment]
@yih-wang Per discussion today with @LHinson and @Mgranfie we have decided to go ahead and merge this as it is.  We recognize that requirements here are a bit fuzzy, and therefore if anyone has a specific need for this feature moving forward, we can update based on those specific requirements. The only change I'd like you to make is to mark this as "Optional" as it is not a default behavior of the list (see my comment).

@serenamarie125 @catrobson FYI.

**Comment by yih-wang** [comment]
@mcarrano Thank you for tracking the task. I've updated the document as requests.

**Comment by mcarrano** [comment]
Thanks for making these updates @yih-wang .  I will go ahead and merge.

**Pull request merged**